rest-client-base/src/main/java/org/jboss/resteasy/microprofile/client/RestClientBuilderImpl.java

```diff
@@ -343,14 +345,48 @@ public RestClientBuilder executorService(ExecutorService executor) {
 
             if (propEngine instanceof ClientHttpEngine) {
                 resteasyClientBuilder.httpEngine((ClientHttpEngine) propEngine);
+            } else {
+                Class httpEngineClazz = null;
+                try {
+                    httpEngineClazz = ClassLoader.getSystemClassLoader().loadClass("org.jboss.resteasy.microprofile.test.client.integration.MPClientHttpEngineConverter");
```

Comment:
This should be moved out of test scope and provided by us (move into a submodule)